src/components/SourcesTree.js

```diff
@@ -54,6 +55,15 @@ let SourcesTree = React.createClass({
       return this.setState({ listItems });
     }
 
+    if (nextProps.hlightSourceURL != this.props.hlightSourceURL) {
+      const hLightItems = getDirectories(
+        nextProps.hlightSourceURL,
+        this.state.sourceTree
+      );
+
+      return this.setState({ hLightItems });
+    }
```

Comment:
I think it's not possible to know which items are expanded in `SourcesTree.js`. Same as `Reveal` , we need to use `setState` and determine which items are expanded in `ManagedTree.js`